source/Pages/compute_on_spider.rst

```diff
@@ -305,7 +318,408 @@ combination that is not available you will receive the following error message:
+
+where the GPU_PARTITION is either ``gpu_v100`` or ``gpu_a100`` depending on which one you are planning to use. The ``--gpus`` flag specifies which type of GPU you want to use and how many, you will get ``N_GPUS`` up to the maximum in the cluster of type ``GPU`` which can be ``v100`` or ``a100``. The compilation and running of code has to be done inside of a singularity container, so start by building a singularity image. More information on singularity on :abbr:`Spider (Symbiotic Platform(s) for Interoperable Data Extraction and Redistribution)` can be found at :ref:`singularity containers <singularity-containers>`. Once the container is available, the program can be run. If container building permissions are not enabled for you on the GPU nodes, please contact us at :ref:`our helpdesk <helpdesk>`.
+
+Next, some short examples for building and running commands are shown. A more in-depth container build procedure is shown :ref:`here <singularity-building>`.
+
+To *interactively* log in to a GPU node run:
+
+.. code-block:: bash
+
+   srun --partition=gpu_v100 --time=00:60:00 --gpus v100:1 --pty bash -i -l
+
+This will open a bash sessions on a machine in the ``gpu_v100`` partition for 60 minutes.
+
+.. tip::
+
+   Asking for more GPUs than the total available on a node does not give an error, your jobs will run on the maximum number.
+
+Simple building example
+=======================
+
+Building can be done as follows:
+
+
+.. code-block:: bash
+
+   singularity build ubuntu.sif docker://ubuntu
+
+In this example, the latest stable version of ubuntu is used (found `here <https://hub.docker.com/_/ubuntu>`_). For running libraries like tensorflow or pytorch or CUDA tools, use the appropriate containers found on the internet. A few links to more resources are given :ref:`here <resources-singularity>`.
+
+After the singularity image has been sucessfully built, the user can enter a shell in the container with:
+
+.. code-block:: bash
+
+   singularity shell --nv ubuntu.sif
+
+In the shell, commands can be run which are executed in the container environment. You can also run a command directly in the container and get the output using ``exec``.
+
+.. code-block:: bash
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+.. WARNING::
+   The ``--nv`` flag is necessary to expose the GPUs on the host the to container.
+
+Here follows an example for running the container in batch mode with a shell script:
+
+.. code-block:: bash
+
+   #!/bin/bash
+
+   #SBATCH -p gpu_v100
+   #SBATCH -G v100:1
+   #SBATCH -e slurm-%j.out
+   #SBATCH -o slurm-%j.out
+
+   singularity exec --nv ubuntu.sif echo "hello world"
+
+The flags ``-e`` and ``-o`` instruct SLURM in which files to write respectively *stderr* and *stdout* of the job. In this case they are both sent to the same file, this is for comparison in the next step. If you now run this shell script on the ``ui-[01-02]`` nodes with ``bash script.sh``, it will result in:
```

Comment:
Not bad perse, but you could leave out 
#SBATCH -e slurm-%j.out
#SBATCH -o slurm-%j.out
We dont use this anywhere else and its not really needed for GPUs and hence may be confusing as why we are suddenly discussing this here

Reply:
I used these two options to get both outputs into the slurm log file, so that the users can see the error for themselves.